tests/assembly/cmse.rs

```diff
@@ -1,6 +1,9 @@
+//@ revisions: hard soft
 //@ assembly-output: emit-asm
-//@ compile-flags: --target thumbv8m.main-none-eabi --crate-type lib -Copt-level=1
-//@ needs-llvm-components: arm
+//@ [hard] compile-flags: --target thumbv8m.main-none-eabihf --crate-type lib -Copt-level=1
+//@ [soft] compile-flags: --target thumbv8m.main-none-eabi --crate-type lib -Copt-level=1
```

Comment:
Question: what's the significance of `opt-level` for this assembly test?

Reply:
mmh not sure anymore I think we just got it from one of the other assembly tests. It seems to work without both `--crate-type lib -Copt-level=1` only the order changes a bit. Should I remove them?

Reply:
I suppose it doesn't matter since we probably want to check that the properties are preserved under opt (though not sure about why specifically `opt-level=1`, what about `2` or `3` or `s` lol)